### PR TITLE
3171 mssql plugin

### PR DESCRIPTION
**Pull request by ilija-lazoroski**
# What does this PR do?

Fixes part of #3171 .

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] Is the TravisCI build passing?
* [x] Was the CHANGELOG.md updated to reflect the changes?
* [x] Was the documentation framework updated to reflect the changes?
* [x] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [x] Added relevant unit tests?
* [x] Do all unit tests pass?
* [x] Do all end-to-end tests pass?
* [x] Any other testing performed?
    > Tested by {Running the Monkey locally with relevant config/running Island/...}
* [x] If applicable, add screenshots or log transcripts of the feature working

### CODE REVIEW

File: monkey/agent_plugins/exploiters/mssql/src/mssql_command_builder.py
Comment (shreyamalviya) [inline monkey/agent_plugins/exploiters/mssql/src/mssql_command_builder.py:1]: No need for this. Just rebase off the other branch.

File: monkey/infection_monkey/exploit/tools/utils.py [hunk 1]
```diff
@@ -19,3 +19,13 @@ def all_udp_ports_are_closed(host: TargetHost, udp_ports: Sequence[NetworkPort])
...
     closed_udp_ports = host.ports_status.udp_ports.closed
     return all([p in closed_udp_ports for p in udp_ports])
+
+
+def any_tcp_port_status_is_unknown(host: TargetHost, tcp_ports: Sequence[NetworkPort]) -> bool:
+    all_host_tcp_ports = host.ports_status.tcp_ports
+    return any([p not in all_host_tcp_ports for p in tcp_ports])
+
+
+def get_open_tcp_ports_by_service(host: TargetHost, service: NetworkService) -> Set[NetworkPort]:
```
[the remark this answers is not in the document]
Comment (shreyamalviya) [reply monkey/infection_monkey/exploit/tools/utils.py:29]: Maybe just `get_open_tcp_ports()` with an optional `network_service` parameter?

File: monkey/agent_plugins/exploiters/mssql/src/plugin.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+            )
+
+        command_builder = partial(
+            build_mssql_command,
+            self._agent_id,
+            servers,
+            current_depth,
+            remote_agent_binary_destination_path=get_agent_dst_path(host),
+            otp_provider=self._otp_provider,
+        )
```
Comment (shreyamalviya) [inline monkey/agent_plugins/exploiters/mssql/src/plugin.py:106]: MSSQL has two commands and hence, no `build_mssql_command()`.

Comment (ilija-lazoroski) [reply monkey/agent_plugins/exploiters/mssql/src/plugin.py:106]: MSSQL seems like a brute-force exploiter but uses HTTP server to download the agent. We may need to write `MSSQLExploiter` after all.

**Review by cakekoa — APPROVED** [review]

**Pull request merged**